presto-hive/src/test/java/io/prestosql/plugin/hive/azure/TestPrestoAzureConfigurationInitializer.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.azure;
+
+import io.airlift.configuration.ConfigurationMetadata;
+import io.airlift.configuration.ConfigurationMetadata.AttributeMetadata;
+import io.airlift.configuration.ConfigurationMetadata.InjectionPointMetaData;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.Sets.combinations;
+import static org.testng.Assert.assertThrows;
+import static org.testng.Assert.fail;
+
+public class TestPrestoAzureConfigurationInitializer
```

Comment:
While clever, this whole class is hard to read, hard to tell what it's doing, and hard to tell that it's correct. Test code should be a simple and clear as possible. Avoiding redundancy is not a goal. The fact that we're basically writing a framework to reflectively call methods says that this is too complicated.

Reply:
Yes, I overdid this a bit.

~~My first thought for how to simplify is to replace `PROPERTY_SETTERS` with a literal map listing the actual setters. That eliminates the reflection, but leaves most of the other complexity and still has helpful test output (including property names).~~

~~My second thought is to remove the property names entirely and replace the whole thing with a setup that has tests that look like this:~~
```
@Test
public void testAbfsAccessKey() {
    testPropertyGroup(HiveAzureConfig::setAbfsAccessKey, HiveAzureConfig::setAbfsStorageAccount);
}
```
~~The class is much easier to follow.~~

On second thought, I'll just go with the second idea. These properties aren't complicated enough that the tests really need to output exactly which properties are set.